pennylane/gradients/param_shift_hessian.py

```diff
@@ -0,0 +1,298 @@
+        # Keep track of tape results already consumed.
+        start = 1 if unshifted_coeffs and f0 is None else 0
+        # Results of the unshifted tape.
+        r0 = results[0] if start == 1 else f0
+
+        for k, ((i, j), s) in enumerate(shapes):
+            res = results[start : start + s]
+            start = start + s
+
+            # Compute the elements of the Hessian as the linear combination of
+            # results and coefficients, barring optimization cases.
+            if j < i:
+                g = hessian[j * h_dim + i]
+            elif s == 0:
+                g = qml.math.zeros(out_dim)
+            else:
+                res = qml.math.stack(res)
+                g = qml.math.tensordot(
+                    res, qml.math.convert_like(hessian_coeffs[k], res), [[0], [0]]
+                )
+                if (i, j) in unshifted_coeffs:
+                    g += unshifted_coeffs[(i, j)] * r0
+
+            hessian.append(g)
+
+        # Reshape the Hessian to have the dimensions of the QNode output on the outside, that is:
+        #         (h_dim, h_dim, out_dim) -> (out_dim, h_dim, h_dim)
+        hessian = qml.math.reshape(qml.math.stack(hessian), (h_dim, h_dim) + out_dim)
+        reordered_axes = list(range(2, len(out_dim) + 2)) + [0, 1]
+        hessian = qml.math.transpose(hessian, axes=reordered_axes)
+
+        return qml.math.squeeze(hessian)
+
+    return hessian_tapes, processing_fn
+
+
+@hessian_transform
+def param_shift_hessian(tape, f0=None):
+    r"""Transform a QNode to compute the parameter-shift Hessian with respect to its trainable
+    parameters.
+
+    Args:
+        tape (pennylane.QNode or .QuantumTape): quantum tape or QNode to differentiate
+        f0 (tensor_like[float] or None): Output of the evaluated input tape. If provided,
+            and the Hessian tapes include the original input tape, the 'f0' value is used
+            instead of evaluating the input tape, reducing the number of device invocations.
+
+    Returns:
+        tensor_like or tuple[list[QuantumTape], function]:
+
+        - If the input is a QNode, a tensor representing the output of the hybrid Hessian matrix
+          of size ``(QNode output dimensions, QNode input dimensions, QNode input dimensions)``
+          is returned. When the keyword ``hybrid=False`` is specified, the purely quantum Hessian
+          matrix is returned instead, with the dimensions
+          ``(QNode output dimensions, number of gate arguments, number of gate arguments)``.
+          The difference between the two accounts for the mapping of QNode arguments
+          to the actual gate arguments, which can include classical computations.
+
+        - If the input is a tape, a tuple containing a list of generated tapes, in addition
+          to a post-processing function to be applied to the evaluated tapes.
+
+    **Examples**
+
+        Applying the Hessian transform to a QNode directly computes the Hessian matrix:
```

Comment:
Looks like there is an indentation error here; this line (and everything below) should be unindented to match the indentation level of `**Examples**`. In ReST, an indent without any other context causes it to be rendered as a quotation (see https://pennylane--1884.org.readthedocs.build/en/1884/code/api/pennylane.gradients.param_shift_hessian.html)

Reply:
Thanks for catching this 😃